src/main/java/com/amazon/ion/impl/macro/MacroCompiler.kt

```diff
@@ -59,7 +58,7 @@ abstract class MacroCompiler(
      * Reads the macro signature, populating parameters in [signature].
      * Caller is responsible for making sure that the reader is positioned on (but not stepped into) the signature sexp.
      */
-    private fun readSignature() {
+    private fun ReaderAdapter.readSignature() {
```

Comment:
This kept the diff to a minimum, but I'm open to other solutions if there's something more idiomatic here.